### ISSUE TITLE
How to pass data to another page when changing route?

### ISSUE THREAD

**Issue by vitobotta** [vuejs/vue-router#947]
HI, I have an app with authentication, so I have a login page which redirects a root page when login is successful. In this case I want to show the message _You are now signed in_ when the root page/component is loaded. I kinda got this working with an alert component included on the root page which is shown when there is a transition from the login page to the root page using beforeRouteEnter on the root component; when beforeRouteEnter is triggered a data attribute on the root component is set to true in turn triggering the display of the alert message.

The problem I have is that if a user is already logged in and tries to load the login page, I want to also redirect to the root page but showing a different message, e.g. _You are already signed in_ (note the _already_). So I was thinking that I could somehow pass data during the redirect from the login component to the root component so that when the root component is loaded it shows the correct message. How would I do this?

I could set a query parameter when redirecting from the login to the root but this way if the user bookmarks the root page, each time they visit that page they will see the _You are already signed in_ message.

Basically what I am trying to do is replicate the _flash message_ on redirect functionality a-la Ruby on Rails.

Any idea? Hope it's clear what I am trying to do. Thanks in advance!

**Comment by fnlctrl** [comment]
Hi, thanks for filling this issue, however the issue tracker is reserved for bug reports and feature requests only, and we encourage you to use our dedicated [gitter](https://gitter.im/vuejs/vue) or [forum](http://forum.vuejs.org/) to ask questions or for discussion. Thanks!

-------------------

Please take a look at [navigation guards](http://router.vuejs.org/en/advanced/navigation-guards.html) and the last example in [route meta fields](http://router.vuejs.org/en/advanced/meta.html)

**Comment by vitobotta** [comment]
Hi @fnlctrl, thanks, will do
